tools/checkissuecomment/checkissuecomment.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main contains linter for todo issue comments.
+package main
+
+import (
+	"bufio"
+	"os"
+	"regexp"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/singlechecker"
+)
+
+var todoIssueRegex = regexp.MustCompile(`(.*?)\/\/\s*TODO https://github\.com/FerretDB/FerretDB/issues/\d+$`)
+
+var analyzer = &analysis.Analyzer{
+	Name: "checkissuecomment",
+	Doc:  "check for TODO comments with issue links",
+	Run:  run,
+}
+
+func main() {
+	singlechecker.Main(analyzer)
+}
+
+// It analyses the presence of TODO issue in code.
+func run(pass *analysis.Pass) (any, error) {
+	for _, file := range pass.Files {
+		fileName := pass.Fset.File(file.Pos()).Name()
+		f, err := os.Open(fileName)
+		if err != nil {
+			return nil, err
+		}
+
+		defer f.Close()
+
+		scanner := bufio.NewScanner(f)
+		lineNumber := 1
+
+		for scanner.Scan() {
+			line := scanner.Text()
+
+			if strings.Contains(line, "// TODO") {
+				if !todoIssueRegex.MatchString(line) {
+					pass.Reportf(file.Pos(), "TODO with missing issue link with proper lint %s", line)
```

Comment:
The current output:

```
./aggregate_documents_compat_test.go:15:1: TODO with missing issue link with proper lint        // TODO Use all providers after fixing $sort problem:  https://github.com/FerretDB/FerretDB/issues/2276.
./aggregate_documents_compat_test.go:15:1: TODO with missing issue link with proper lint        // TODO Merge the current function with TestAggregateCompatGroupExpressionDottedFields
./aggregate_documents_compat_test.go:15:1: TODO with missing issue link with proper lint                // TODO: handle $sum of doubles near max precision.
./aggregate_documents_compat_test.go:15:1: TODO with missing issue link with proper lint                // TODO: https://github.com/FerretDB/FerretDB/issues/2616
./aggregate_documents_compat_test.go:15:1: TODO with missing issue link with proper lint                // TODO: https://github.com/FerretDB/FerretDB/issues/2617
./aggregate_stats_compat_test.go:15:1: TODO with missing issue link with proper lint                                    // TODO Check the returned values when possible: https://github.com/FerretDB/FerretDB/issues/2349
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   // TODO Set better expected results https://github.com/FerretDB/FerretDB/issues/1771
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   // TODO Set better expected results https://github.com/FerretDB/FerretDB/issues/1771
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   // TODO Set better expected results https://github.com/FerretDB/FerretDB/issues/1771
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   // TODO assert.Empty(t, doc.Keys())
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   assert.EqualValues(t, float64(1), doc.Remove("scaleFactor")) // TODO use assert.Equal https://github.com/FerretDB/FerretDB/issues/727
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   // TODO assert.Empty(t, doc.Keys())
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   // TODO assert.Empty(t, doc.Keys())
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   assert.EqualValues(t, float64(1000), doc.Remove("scaleFactor")) // TODO use assert.Equal https://github.com/FerretDB/FerretDB/issues/727
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   // TODO assert.Empty(t, doc.Keys())
./commands_administration_test.go:15:1: TODO with missing issue link with proper lint   // TODO rewrite using teststress.Stress
./count_command_compat_test.go:15:1: TODO with missing issue link with proper lint      // TODO Use read-only user. https://github.com/FerretDB/FerretDB/issues/1025
./count_compat_test.go:15:1: TODO with missing issue link with proper lint      // TODO Use read-only user. https://github.com/FerretDB/FerretDB/issues/1025
./create_test.go:15:1: TODO with missing issue link with proper lint    // TODO rewrite using teststress.Stress
./create_test.go:15:1: TODO with missing issue link with proper lint    // TODO rewrite using teststress.Stress
./create_test.go:15:1: TODO with missing issue link with proper lint    // TODO rewrite using teststress.Stress
./create_test.go:15:1: TODO with missing issue link with proper lint    // TODO rewrite using teststress.Stress
./distinct_compat_test.go:15:1: TODO with missing issue link with proper lint   // TODO Use read-only user. https://github.com/FerretDB/FerretDB/issues/1025
./findandmodify_compat_test.go:15:1: TODO with missing issue link with proper lint // TODO Add more tests for sort: https://github.com/FerretDB/FerretDB/issues/2168
./findandmodify_test.go:15:1: TODO with missing issue link with proper lint                     // TODO: add document comparison here. Skip _id check as it always would different.
./getmore_test.go:15:1: TODO with missing issue link with proper lint                   // TODO: 16MB batchSize limit https://github.com/FerretDB/FerretDB/issues/2824
./getmore_test.go:15:1: TODO with missing issue link with proper lint                   // TODO: 16MB batchSize limit https://github.com/FerretDB/FerretDB/issues/2824
./query_array_test.go:15:1: TODO with missing issue link with proper lint                       // TODO: move to compat https://github.com/FerretDB/FerretDB/issues/1540
./query_bitwise_test.go:15:1: TODO with missing issue link with proper lint     // TODO: move to compat https://github.com/FerretDB/FerretDB/issues/508
./query_bitwise_test.go:15:1: TODO with missing issue link with proper lint     // TODO: move to compat https://github.com/FerretDB/FerretDB/issues/508
./query_bitwise_test.go:15:1: TODO with missing issue link with proper lint     // TODO: move to compat https://github.com/FerretDB/FerretDB/issues/508
./query_bitwise_test.go:15:1: TODO with missing issue link with proper lint     // TODO: move to compat https://github.com/FerretDB/FerretDB/issues/508
./query_command_compat_test.go:15:1: TODO with missing issue link with proper lint      // TODO Use read-only user. https://github.com/FerretDB/FerretDB/issues/1025
./query_compat_test.go:15:1: TODO with missing issue link with proper lint      // TODO Use read-only user. https://github.com/FerretDB/FerretDB/issues/1025
./query_compat_test.go:15:1: TODO with missing issue link with proper lint              // TODO: https://github.com/FerretDB/FerretDB/issues/2618
./query_evaluation_test.go:15:1: TODO with missing issue link with proper lint  // TODO: move to compat https://github.com/FerretDB/FerretDB/issues/1576
./update_compat_test.go:15:1: TODO with missing issue link with proper lint                                                     // TODO replace with UpdateMany/ReplaceMany

```

It seems that we handle proper cases, although the message seems a bit irrelevant. I'd suggest to create a message like: "TODO comments must satisfy the pattern: // TODO <issue url>. Also, let's trim the heading spaces/tabs before // TODO in the message